libcnb/src/build.rs

```diff
@@ -95,18 +104,297 @@ impl<B: Buildpack + ?Sized> BuildContext<B> {
     ///     }
     /// }
     /// ```
-    pub fn handle_layer<L: Layer<Buildpack = B>>(
+    #[deprecated = "The Layer trait API was replaced by a struct based API. Use `cached_layer` and `uncached_layer`."]
+    #[allow(deprecated)]
+    pub fn handle_layer<L: crate::layer::Layer<Buildpack = B>>(
         &self,
         layer_name: LayerName,
         layer: L,
-    ) -> crate::Result<LayerData<L::Metadata>, B::Error> {
-        crate::layer::handle_layer(self, layer_name, layer).map_err(|error| match error {
-            HandleLayerErrorOrBuildpackError::HandleLayerError(e) => {
-                crate::Error::HandleLayerError(e)
+    ) -> crate::Result<crate::layer::LayerData<L::Metadata>, B::Error> {
+        crate::layer::trait_api::handling::handle_layer(self, layer_name, layer).map_err(|error| {
+            match error {
+                LayerErrorOrBuildpackError::LayerError(e) => crate::Error::LayerError(e),
+                LayerErrorOrBuildpackError::BuildpackError(e) => crate::Error::BuildpackError(e),
             }
-            HandleLayerErrorOrBuildpackError::BuildpackError(e) => crate::Error::BuildpackError(e),
         })
     }
+
+    /// Creates a cached layer, potentially re-using a previously cached version.
+    ///
+    /// Buildpack code uses this function to create a cached layer and will get back a reference to
+    /// the layer directory on disk. Intricacies of the CNB spec are automatically handled such as
+    /// the maintenance of TOML files. Buildpack code can also specify a callback for cached layer
+    /// invalidation.
+    ///
+    /// Users of this function pass in a [`CachedLayerDefinition`] that describes the desired layer
+    /// and the returned `LayerRef` can then be used to modify the layer like any other path. This
```

Comment:
Make doc reference clickable

```suggestion
    /// and the returned [`LayerRef`] can then be used to modify the layer like any other path. This
```

Make doc clickable

Reply:
👍🏻